src/nyc_trees/apps/survey/views.py

```diff
@@ -189,9 +195,63 @@ def start_survey_from_event(request, event_slug):
     }
 
 
-def submit_survey(request, blockface_id):
-    # TODO: implement
-    pass
+def submit_survey(request):
+    return _create_survey_and_trees(request)
+
+def submit_survey_from_event(request, event_slug):
+    event = get_object_or_404(Event, group=request.group, slug=event_slug)
+    if not user_is_checked_in_to_event(request.user, event):
+        return HttpResponseForbidden('User not checked-in to this event')
+
+    return _create_survey_and_trees(request, event)
+
+
+@transaction.atomic
+def _create_survey_and_trees(request, event=None):
+    """
+    Creates survey and trees from JSON body, where k1... are model attrs: {
+        survey: { k1:v1, ... },
+        trees: [
+            { k1:v1, ...},
+            ...
+        ]
+    }
+    trees.problems should be a list of problem codes -- ["Stones", "Sneakers"]
+    """
+    data = json.loads(request.body)
+    survey_data = data['survey']
+    tree_list = data.get('trees', [])
+
+    survey = Survey(user=request.user, **survey_data)
+    blockface = survey.blockface
+
+    if event:
+        territory = _get_territory(blockface)
+        if territory is None or territory.group_id != event.group_id:
+            return HttpResponseForbidden(
+                "Blockface is not in group's territory.")
+    else:
+        if not _query_reservation(request.user, blockface.id).exists():
+            return HttpResponseForbidden('You have not reserved this blockface.')
+
+    if survey.has_trees and len(tree_list) == 0:
+        return HttpResponseBadRequest('Trees expected but absent')
+    if not survey.has_trees and len(tree_list) > 0:
+        return HttpResponseBadRequest('Trees not expected but present')
```

Comment:
Note the UI has a "No trees?" button that needs special handling. I put in these checks to guard against mistakes in that code, so I'd rather leave them in. The client code seems straightforward since `has_trees` will always be true for one branch and always false for the other.